File: 25.R
# 1000-digit Fibonacci number

Args   <- commandArgs()
digits <- ifelse(is.na(Args[6]), 1000, as.numeric(Args[6]))

ordinal <- 1
fib1 <- c(1)
fib2 <- c(1)

while ( length(fib1) < digits ) {
    ordinal <- ordinal + 1
    carry <- 0
    sum <- vector()
    for ( digit in seq(length(fib2)) ) {
        fib1_digit <- ifelse(is.na(fib1[digit]), 0, fib1[digit])
        digit_sum <- fib2[digit] + fib1_digit + carry
        last_digit <- as.integer(substr(digit_sum, nchar(digit_sum),
                                        nchar(digit_sum)))
        sum <- c(sum, last_digit)
        if ( digit_sum >= 10 ) {
            carry <- as.integer(substr(digit_sum, 1, nchar(digit_sum)-1))
        } else {
            carry <- 0
        }
    }
    if ( carry > 0 ) {
        carry_digits <- rev(as.integer(strsplit(as.character(carry),
                                                split=NULL)[[1]]))
        sum <- c(sum, carry_digits)
    }
    fib1 <- fib2
    fib2 <- sum
}

cat(ordinal, fill=TRUE)

File: 25.js
#!/usr/bin/env node

// 1000-digit Fibonacci number

var program = require('commander');

program
    .version('0.1.0')
    .description('1000-digit Fibonacci number')
    .option('-d, --digits <int>', 'The number of digits', Number, 1000)
    .parse(process.argv);

var ordinal = 1;
var fib1 = [1];
var fib2 = [1];
while (fib1.length < program.digits) {
    ordinal++;
    var carry = 0;
    var fib3 = [];
    for (var i = 0; i < fib2.length; i++) {
        var sum = fib2[i] + (fib1[i] || 0) + (carry || 0);
        var last_digit_of_sum = parseInt(sum.toString().slice(-1));
        fib3.push(last_digit_of_sum);
        carry = parseInt(sum.toString().slice(0, -1));
    }
    if (carry) {
        var carry_digits = carry.toString().split("").reverse();
        for (var i = 0; i < carry_digits.length; i++) {
            fib3.push(parseInt(carry_digits[i]));
        }
    }
    fib1 = fib2;
    fib2 = fib3;
}

console.log(ordinal);
